File: src/main/java/it/xpug/kata/birthday_greetings/Employee.kt
package it.xpug.kata.birthday_greetings

class Employee(
    val firstName: String,
    val lastName: String,
    birthDate: String,
    val email: String
) {

    private val xdate = XDate(birthDate)

    fun isBirthday(today: XDate): Boolean {
        return today.isSameDay(xdate)
    }

    override fun toString(): String {
        return "Employee $firstName $lastName <$email> born $xdate"
    }
}

File: csharp/BirthdayGreetings/Employee.cs
namespace BirthdayGreetings
{
    public class Employee
    {
        public string FirstName { get; }
        public string LastName { get; }
        private XDate xdate;
        public string Email { get; }

        public Employee(string firstName, string lastName, string birthDate, string email)
        {
            FirstName = firstName;
            LastName = lastName;
            xdate = new XDate(birthDate);
            Email = email;
        }

        public bool IsBirthday(XDate today)
        {
            return today.IsSameDay(xdate);
        }

        public override string ToString()
        {
            return $"Employee {FirstName} {LastName} <{Email}> born {xdate}";
        }
    }
} 